File: app/src/test/java/ar/com/wolox/android/example/ui/example/ExamplePresenterTest.java
package ar.com.wolox.android.example.ui.example;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

import ar.com.wolox.android.example.utils.UserSession;

import org.junit.Before;
import org.junit.Test;

public class ExamplePresenterTest {

    private IExampleView mExampleView;
    private ExamplePresenter mExamplePresenter;
    private UserSession mUserSession;

    @Before
    public void createInstances() {
        mExampleView = mock(IExampleView.class);
        mUserSession = mock(UserSession.class);
        mExamplePresenter = new ExamplePresenter(mUserSession);
    }

    @Test
    public void usernameIsStored() {
        mExamplePresenter.attachView(mExampleView);
        mExamplePresenter.storeUsername("Test");
        verify(mUserSession, times(1)).setUsername("Test");
    }

    @Test
    public void storeUsernameUpdatesView() {
        mExamplePresenter.attachView(mExampleView);
        mExamplePresenter.storeUsername("Test");
        verify(mExampleView, times(1)).onUsernameSaved();
    }

}

File: app/src/test/java/ar/com/wolox/android/example/ui/example/ExamplePresenterTest.kt
package ar.com.wolox.android.example.ui.example

import ar.com.wolox.android.example.utils.UserSession
import ar.com.wolox.wolmo.core.tests.WolmoPresenterTest
import com.nhaarman.mockitokotlin2.times
import com.nhaarman.mockitokotlin2.verify
import org.junit.Test
import org.mockito.Mock

class ExamplePresenterTest : WolmoPresenterTest<ExampleView, ExamplePresenter>() {

    @Mock
    lateinit var userSession: UserSession

    override fun getPresenterInstance() = ExamplePresenter(userSession)

    @Test
    fun `given an user and a color when login button is clicked then user session should be updated`() {

        // GIVEN
        val user = "Test"
        val color = "_"

        // WHEN
        presenter.onLoginButtonClicked(user, color)

        // THEN
        verify(userSession, times(1)).username = user
    }

    @Test
    fun `given an user and a color when login button is clicked then view should go to viewpager`() {

        // GIVEN
        val user = "Test"
        val color = "blue"

        // WHEN
        presenter.onLoginButtonClicked(user, color)

        // THEN
        verify(view, times(1)).goToViewPager(color)
    }
}
